numpy/core/src/umath/ufunc_object.c

```diff
@@ -4043,45 +4125,41 @@ _find_array_wrap(PyObject *args, PyObject *kwds,
      */
 handle_out:
     nargs = PyTuple_GET_SIZE(args);
-    for (i = 0; i < nout; i++) {
-        int j = nin + i;
-        int incref = 1;
-        output_wrap[i] = wrap;
-        obj = NULL;
-        if (j < nargs) {
-            obj = PyTuple_GET_ITEM(args, j);
-            /* Output argument one may also be in a keyword argument */
-            if (i == 0 && obj == Py_None && kwds != NULL) {
-                obj = PyDict_GetItem(kwds, npy_um_str_out);
-            }
+
+    if (nin == nargs && kwds != NULL &&
+                        (obj = PyDict_GetItem(kwds, npy_um_str_out))) {
```

Comment:
Here also I would suggest taking the assignment out of the `if` statement and then simply doing another `if`.